misc/bazel/pkg.bzl

```diff
@@ -351,25 +352,42 @@ def _codeql_pack_install(name, srcs, install_dest = None, build_file_label = Non
             visibility = ["//visibility:private"],
         )
         build_file_label = internal("build-file")
-
+    data = [
+        internal("script"),
+        internal("zip-manifest"),
+        Label("//misc/ripunzip"),
+    ] + ([build_file_label] if build_file_label else [])
+    args = [
+        "--pkg-install-script=$(rlocationpath %s)" % internal("script"),
+        "--ripunzip=$(rlocationpath %s)" % Label("//misc/ripunzip"),
+        "--zip-manifest=$(rlocationpath %s)" % internal("zip-manifest"),
+    ] + ([
+        "--build-file=$(rlocationpath %s)" % build_file_label,
+    ] if build_file_label else []) + (
+        ["--destdir", "\"%s\"" % install_dest] if install_dest else []
+    )
     py_binary(
         name = name,
         srcs = [Label("//misc/bazel/internal:install.py")],
         main = Label("//misc/bazel/internal:install.py"),
-        data = [
-            internal("script"),
-            internal("zip-manifest"),
-            Label("//misc/ripunzip"),
-        ] + ([build_file_label] if build_file_label else []),
         deps = ["@rules_python//python/runfiles"],
-        args = [
-                   "--pkg-install-script=$(rlocationpath %s)" % internal("script"),
-                   "--ripunzip=$(rlocationpath %s)" % Label("//misc/ripunzip"),
-                   "--zip-manifest=$(rlocationpath %s)" % internal("zip-manifest"),
-               ] + ([
-                   "--build-file=$(rlocationpath %s)" % build_file_label,
-               ] if build_file_label else []) +
-               (["--destdir", "\"%s\"" % install_dest] if install_dest else []),
+        data = data,
+        args = args,
+    )
+
+    # this hack is meant to be an optimization when using install for tests, where
+    # the install step is skipped if nothing changed. If the installation directory
+    # is somehow messed up, `bazel run` can be used to force install
+    native_test(
+        name = internal("as", "test"),
+        src = name,
+        tags = [
+            "manual",  # avoid having this picked up by `...`, `:all` or `:*`
+            "local",  # make sure installation does not run sandboxed
+        ],
+        data = data,
+        args = args,
+        size = "small",
```

Comment:
the timeout only covers the install part, right? Which under all circumstances ought to finish in 60 seconds, so this is appropriate.

Reply:
indeed!